pkg/resource/replication_group/custom_set_output.go

```diff
@@ -249,3 +264,31 @@ func (rm *resourceManager) provideEvents(
 	}
 	return events, nil
 }
+
+// to be called upon a successful creation or modification (but not read/describe),
+// setLastRequestedLogDeliveryConfigurations copies desired.Spec.LogDeliveryConfigurations
+// into the annotations of the object.
+// r represents desired, ko represents latest
+func (rm *resourceManager) setLastRequestedLogDeliveryConfigurations(
+	r *resource,
+	ko *svcapitypes.ReplicationGroup,
+) {
+	//TODO: if in the future we need to set other annotations fields, ensure all get persisted
+	// as opposed to each making a copy of desired and only modifying one field
```

Comment:
You could modify this method signature to accept the annotations map, and modify only this key. Then in the parent functions, it from object meta and persist it back afterwards?

Reply:
Thanks for the suggestion! The latest version now uses this model